packages/opencensus-instrumentation-mongodb/src/mongodb.ts

```diff
@@ -45,109 +50,110 @@ export class MongoDBPlugin extends classes.BasePlugin {
       this.logger.debug('patching mongodb-core.Server.prototype.command');
       shimmer.wrap(
-          moduleExports.Server.prototype, 'command', this.patchCommand());
+          moduleExports.Server.prototype, 'command' as never,
+          this.patchCommand());
       this.logger.debug(
           'patching mongodb-core.Server.prototype functions:', this.SERVER_FNS);
       shimmer.massWrap(
-          [moduleExports.Server.prototype], this.SERVER_FNS, this.patchQuery());
+          [moduleExports.Server.prototype], this.SERVER_FNS as never[],
+          this.patchQuery());
     }
 
     if (moduleExports.Cursor) {
       this.logger.debug(
           'patching mongodb-core.Cursor.prototype functions:',
           this.CURSOR_FNS_FIRST);
       shimmer.massWrap(
-          [moduleExports.Cursor.prototype], this.CURSOR_FNS_FIRST,
+          [moduleExports.Cursor.prototype], this.CURSOR_FNS_FIRST as never[],
           this.patchCursor());
     }
 
     return moduleExports;
   }
 
-  /** Unpatches all HTTP patched function. */
+  /** Unpatches all MongoDB patched functions. */
   applyUnpatch(): void {
     shimmer.unwrap(this.moduleExports.Server.prototype, 'command');
     shimmer.massUnwrap(this.moduleExports.Server.prototype, this.SERVER_FNS);
     shimmer.massUnwrap(
         this.moduleExports.Cursor.prototype, this.CURSOR_FNS_FIRST);
   }
 
-  patchCommand() {
+  /** Creates spans for Command operations */
+  private patchCommand() {
     const plugin = this;
     return (original: types.Func<mongodb.Server>) => {
-      // tslint:disable-next-line:no-any
-      return function(this: mongodb.Server, ns: string, command: any):
-          mongodb.Server {
-            let span: types.Span;
-            const index = arguments.length - 1;
-            const resultHandler = arguments[index];
-
-            if (plugin.tracer.currentRootSpan && arguments.length > 0 &&
-                typeof resultHandler === 'function') {
-              let type: string;
-              if (command.createIndexes) {
-                type = 'createIndexes';
-              } else if (command.findandmodify) {
-                type = 'findAndModify';
-              } else if (command.ismaster) {
-                type = 'ismaster';
-              } else if (command.count) {
-                type = 'count';
-              } else {
-                type = 'command';
-              }
-
-              span = plugin.tracer.startChildSpan(
-                  ns + '.' + type, plugin.SPAN_MONGODB_QUERY_TYPE);
-              arguments[index] = plugin.patchEnd(span, resultHandler);
-            }
-
-            return original.apply(this, arguments);
-          };
+      return function(
+                 // tslint:disable-next-line:no-any
+                 this: mongodb.Server, ns: string, command: any,
+                 preferences: object, resultHandler: Function): mongodb.Server {
+        if (plugin.tracer.currentRootSpan && arguments.length > 0 &&
```

Comment:
This will only work if there is guaranteed to be exactly 4 arguments. My comment earlier was on whether we could do something like

```js
function(this, ...args: any[])
```

and using `args` in place of `arguments`. Not sure if syntactically that will work -- if it doesn't, I think the previous approach (`arguments.length - 1`) is better.

Reply:
Changed